src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -63,7 +63,17 @@ def is_dxlink(x):
     Returns whether *x* appears to be a DNAnexus link (is a dict with
     key ``"$dnanexus_link"``) with a referenced data object.
     '''
-    return isinstance(x, dict) and '$dnanexus_link' in x and (isinstance(x['$dnanexus_link'], basestring) or isinstance(x['$dnanexus_link'], dict) and 'id' in x['$dnanexus_link'])
+    if not isinstance(x, dict):
+        return False
+    if '$dnanexus_link' not in x:
```

Comment:
On second thought, I've reverted that change. Let's keep these clauses separate to avoid any possible confusion from the predicate ordering (leading `not` or the need to use parentheses for grouping).